### ISSUE TITLE
USB print

### ISSUE THREAD

**Issue by safanmomin** [InteractiveObject/StarIOPlugin#10]
I have a Star Micronics USB printer. Does this plugin Supports USB printing?

**Comment by younes200** [comment]
You should re-test with @ttisch PR

**Comment by safanmomin** [comment]
@ttisch can you help?

**Comment by ttisch** [comment]
@safanmomin  I did not try it via USB nor LAN. But I was facing problems when I started 'portDiscovery' with parameter 'All' when connecting via Bluetooth:
`.portDiscovery('All', function(error, printerList){ // NOT WORKING`

I had to specify port name explicitly:
`.portDiscovery('BluetoothLE', function(error, printerList){ //WORKING`

That's all I can say. Hope it helps...

**Comment by safanmomin** [comment]
Thanks for replying @ttisch. For me 'portDiscovery' with parameter 'LAN' is working and it connects with the printer too put the function 'printReceipt' doesn't work.

**Comment by ttisch** [comment]
@safanmomin Ok, now I remember I had issues too. I think README should be updated regarding parameter 'portName' in 'printReceipt'.

I did it this way: (code simplified) Maybe this helps.

```
window.plugins.starPrinter.portDiscovery('BluetoothLE', function (error, result) {
    $scope.printers = result;
});

window.plugins.starPrinter.printReceipt($scope.printers[0].portName, 'Test123', function(error, result){
    if (error) {
        console.error(error);
   } else {
       console.log("printReceipt finished");
   }
});
```

**Comment by safanmomin** [comment]
I logged `printers[0].portName` and returns undefined. I logged the results and it returns the following JSON:

`[{"macAddress":"mac address of printer 1","name":"name of printer 1"},{"macAddress":"mac address of printer 2","name":"name of printer 2"}]`

No `portName` property in the JSON.

**Comment by offlineprogrammer** [comment]
I get printers[0].portName as undefined.....

**Comment by sourcer18** [comment]
Correct, appears indefinite,@offlineprogrammer  if you find the solution, please communicate, just as I will

**Comment by tapndine** [comment]
Has anyone gotten this working yet? I can confirm that I have a connection to a LAN printer and it responds correctly if I open the cover and such, but sending the printReceipt command returns as "Printed" however the printer doesnt actually do anything.
